search/controllers/simple/__init__.py

```diff
@@ -72,7 +72,7 @@ def search(request_params: dict) -> Response:
 
     # Fall back to form-based search.
     form = SimpleSearchForm(request_params)
-    q = None
+    q: Query
```

Comment:
See line 98: we do actually set this to None; should it be ``Optional[Query]``?

Reply:
Here's some fun:

```
>>> a: int
>>> a
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'a' is not defined
>>> a = None
>>> a
>>> 
```

submitting a patch for `Optional[Query]` then merging.

Reply:
Patch will be on all controllers with that type of error.